docs/src/userguide/language_basics.rst

```diff
@@ -119,40 +187,88 @@ You can read more about them in :ref:`extension-types`.
-``long long`` as well as their ``unsigned`` versions, e.g. ``unsigned int``.
-The special ``bint`` type is used for C boolean values (``int`` with 0/non-0
-values for False/True) and ``Py_ssize_t`` for (signed) sizes of Python
-containers.
+.. tabs::
+
+    .. group-tab:: Pure Python
+
+        Cython supports annotaions for C types, including pointers.  It provides
+        all the standard C types, namely ``char``, ``short``, ``int``, ``long``,
+        ``long long`` as well as their ``unsigned`` versions, e.g. ``uint``.
+        The special ``bint`` type is used for C boolean values (``int`` with 0/non-0
+        values for False/True) and ``Py_ssize_t`` for (signed) sizes of Python
+        containers.
+
+        Pointer types are constructed by appending a ``p_`` to the base type
+        they point to, e.g. ``pp_int`` for a pointer to a pointer to a C int.
+        Further pointer types can be constructed with ``pointer()`` function.
+        Arrays use the normal C array syntax, e.g. ``int[10]``, and the size must be known
+        at compile time for stack allocated arrays. Cython doesn't support variable length arrays from C99.
+        Note that Cython uses array access for pointer dereferencing.
+
+    .. group-tab:: Cython
 
-Pointer types are constructed as in C, by appending a ``*`` to the base type
-they point to, e.g. ``int**`` for a pointer to a pointer to a C int.
-Arrays use the normal C array syntax, e.g. ``int[10]``, and the size must be known
-at compile time for stack allocated arrays. Cython doesn't support variable length arrays from C99.
-Note that Cython uses array access for pointer dereferencing, as ``*x`` is not valid Python syntax,
-whereas ``x[0]`` is.
+        Cython language uses the normal C syntax for C types, including pointers.  It provides
+        all the standard C types, namely ``char``, ``short``, ``int``, ``long``,
+        ``long long`` as well as their ``unsigned`` versions, e.g. ``unsigned int``.
+        The special ``bint`` type is used for C boolean values (``int`` with 0/non-0
+        values for False/True) and ``Py_ssize_t`` for (signed) sizes of Python
+        containers.
+
+        Pointer types are constructed as in C, by appending a ``*`` to the base type
+        they point to, e.g. ``int**`` for a pointer to a pointer to a C int.
+        Arrays use the normal C array syntax, e.g. ``int[10]``, and the size must be known
+        at compile time for stack allocated arrays. Cython doesn't support variable length arrays from C99.
+        Note that Cython uses array access for pointer dereferencing, as ``*x`` is not valid Python syntax,
+        whereas ``x[0]`` is.
 
 Also, the Python types ``list``, ``dict``, ``tuple``, etc. may be used for
 static typing, as well as any user defined :ref:`extension-types`.
-For example::
+For example
+
+.. tabs::
+
+    .. group-tab:: Pure Python
+
+        .. code-block:: python
+
+            def main():
+                foo: list = []
```

Comment:
Must he module-level, please remove main

Reply:
@0dminnimda to be honest I don't understand. @scoder mentioned that annotated variables should be moved to function since Cython does not support annotated global variables:

https://github.com/cython/cython/pull/4242#discussion_r660033760
https://github.com/cython/cython/pull/4242#pullrequestreview-694259168

Hence, I put all code to `main()` function. Or better idea is to use `declare()`? I am not sure what is better :-(

Reply:
Not sure if `main()` is a good name, but it's definitely good enough for now and easy to change later. Let's stick with it.

Reply:
> Or better idea is to use declare()?

that's the other way of doing this